spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/datastore/health/DatastoreHealthIndicatorAutoConfiguration.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.datastore.health;
+
+import com.google.cloud.datastore.Datastore;
+
+import org.springframework.boot.actuate.autoconfigure.health.ConditionalOnEnabledHealthIndicator;
+import org.springframework.boot.actuate.autoconfigure.health.HealthIndicatorAutoConfiguration;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.cloud.gcp.autoconfigure.datastore.GcpDatastoreAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * {@link HealthIndicatorAutoConfiguration Auto-configuration} * for
+ * {@link DatastoreHealthIndicator}.
+ *
+ * @author Raghavan N S
+ * @author Srinivasa Meenavalli
+ * @author Mike Eltsufin
+ *
+ * @since 1.2
+ */
+@Configuration
+@ConditionalOnClass(Datastore.class)
+@ConditionalOnBean(Datastore.class)
+@ConditionalOnEnabledHealthIndicator("datastore")
```

Comment:
I'm not sure that such a simple, generic *indicator* is really enough to distinguish GCP Datastore from any others.
What is the problem to have really more explicit name, e.g. `gcp.datastore`, at least?..

Reply:
This would only be a thing if someone is using Spring Cloud GCP, and there is only one "Datastore" in GCP. We can also change it to the more complete official name "cloud-datastore". WDYT?